content/client-lib-development-guide/features.textile

```diff
@@ -1242,6 +1258,31 @@ h4. TokenRequest
 * @(TE4)@ @ttl@ attribute represents time to live (expiry) of this token in milliseconds
 * @(TE6)@ @TokenRequest::fromJson@ static factory method that accepts either a @JsonObject@ or a string (which should be parsed as a JSON string), and returns a new @TokenRequest@. Statically typed languages (that expect the @authCallback@ to result in a typed @TokenRequest@ object, rather than, say, a hashmap) must implement this; others may at their discretion.
 
+h4. TokenRevocationRequest
+
+* @(TF1)@ @TokenRevocationRequest@ is a type containing the token revocation request details sent to the "REST revokeTokens endpoint":/rest-api/#revokeTokens
+* @(TF2)@ @TokenRevocationRequest#key@ attribute is a string representing key to revoke (appId.Key).
+* @(TF3)@ @TokenRevocationRequest#targets@ is an array of @TokenRevocationTarget@s to revoke. For request to be valid, it must contain at least one target. A test must exist to ensure there is at least one target in the request before it is sent.
+* @(TF4)@ @TokenRevocationRequest#issuedBefore@ is an optional timestamp (in milliseconds since the epoch) to revoke tokens issued before this time.
+* @(TF5)@ @TokenRevocationRequest#allowReauthMargin@ is an optional boolean that permits a token renewal cycle to take place without needing established connections to be dropped. Default for this value is false if not provided.
+
+h4. TokenRevocationTarget
+
+* @(TL1)@ @TokenRevocationTarget@ is a type representing a target for "REST revokeTokens endpoint":/rest-api/#revokeTokens.
+* @(TL2)@ @TokenRevocationTarget#key@ is a string containing the target key. Target key can be "clientId","revocationKey" or "channel". Where possible, clients should predefine those strings (enums for example) to prevent customers passing an invalid value.
```

Comment:
Should we be `enum`ing this?  This would mean if we want to support new targets, we have to roll out updates to all clients.  Would it not be better in fact to keep this as strings and instead provide advice in the inline docs for developers to reduce overhead in maintaining this?  I think overall we're erring towards client specs that don't need to change when server-side functionality expands.

Reply:
In addition, if we later changed the response for `RevokedToken` below, and we had a new target type, we'd break the SDK as it would end up with a null value / error trying to create the `RevokedToken` object.

Reply:
I think enum'ing will add some clarity and safety but as you mentioned it will also make it incompatible with any new changes made to API. I wasn't particularly sure what's the guidance around how we interpret such API objects and suggested enuming based on supported attributes here
https://ably.com/docs/core-features/authentication#target-specifiers
Now you pointed out I will update this so it's more permissive

Reply:
Removed the wording on enuming key 6868b0b3971edbc99fd4262749e702caafbc77f6